core/src/main/java/org/apache/accumulo/core/metadata/schema/UnSplittableMetadata.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.metadata.schema;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.accumulo.core.util.LazySingletons.GSON;
+
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.accumulo.core.metadata.StoredTabletFile;
+import org.apache.accumulo.core.util.json.ByteArrayToBase64TypeAdapter;
+
+import com.google.common.base.Preconditions;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hashing;
+import com.google.gson.Gson;
+
+public class UnSplittableMetadata {
+
+  private static final Gson GSON = ByteArrayToBase64TypeAdapter.createBase64Gson();
+
+  private final long splitThreshold;
+  private final long maxEndRowSize;
+  private final int maxFilesToOpen;
+  private final HashCode filesHash;
```

Comment:
Thinking it may simplify the code in this class if all we store is a hash.  Something like the following.  Would remove the need for the json code.
```suggestion
  // stores a base64 encoded murmur 128 bit hash splitThreshold,maxEndRowSize,maxFilesToOpen, set of files.
  private final String hashOfSplitParameters;
```

Reply:
This would definitely simply things but can you think of a reason we may need to read the original values? (maybe logging?) This change is probably fine as I don't think we need to query those stored values individually again but figure I'd double check before changing it. We could also log the values that are used to create the hash (maybe as trace) so we could access it there.

Reply:
> can you think of a reason we may need to read the original values?

The only case I can think of currently is debugging a problem in the code and I think for that case your suggestion to log covers that.

> We could also log the values that are used to create the hash (maybe as trace) so we could access it there.

I think logging the inputs to the hash and the resulting hash would be very useful.  Then if there is a problem can search the logs for the hash.

Reply:
Changed in in 6c3358b5783b19ec6d73c7f7a121d0b0493c4333